apps/microsoft-teams/frontend/src/hooks/useGetContentTypes.ts

```diff
@@ -18,17 +19,41 @@ const useGetContentTypes = () => {
   const getAllContentTypes = useCallback(async () => {
     try {
       setLoading(true);
-      // TODO: Implement pagination for content types
-      const contentTypesResponse = await sdk.cma.contentType.getMany({});
-      setAllContentTypes(contentTypesResponse.items || []);
+
+      const contentTypesSuperset: ContentTypeProps[] = [];
+      let skip = 0;
+      let needToFetchNextBatch = true;
+
+      while (needToFetchNextBatch) {
+        const contentTypesResponse = await sdk.cma.contentType.getMany({
+          query: {
+            limit: CONTENT_TYPE_LIMIT,
+            skip,
+            order: 'name',
+          },
+        });
+
+        const totalNumOfContentTypes = contentTypesResponse.total;
+        contentTypesSuperset.push(...contentTypesResponse.items);
+
+        if (contentTypesSuperset.length >= totalNumOfContentTypes) {
+          // we have fetched all possible content types, exit while
+          needToFetchNextBatch = false;
+        } else {
+          // we need to fetch the next batch of content types, re-execute while
+          skip += CONTENT_TYPE_LIMIT;
```

Comment:
Just a quick callout, I would check to see if we're skipping the 100th index every call. 

Depending on how the skip is setup we could be getting 0 - 99, then on the next call getting 101-200, etc. Or this could be 100% fine and a non issue

Reply:
Good call out, I tested this out by setting the `CONTENT_TYPE_LIMIT` to 2 and adding/subtracting 1 on this line.
 
`skip += CONTENT_TYPE_LIMIT - 1` will fetch records `[1, 2] , [2, 3] , [3, 4]`
`skip += CONTENT_TYPE_LIMIT + 1` will fetch records `[1, 2] , [4, 5], [7]`

So we should be good with `skip += CONTENT_TYPE_LIMIT`, which I varifyed fetches records `[1,2] [3,4] [5, 6] [7]` in batches of 2, it works similarly for batches of 3..4...n